activesupport/test/broadcast_logger_test.rb

```diff
@@ -307,6 +307,72 @@ def info(msg, &block)
       assert_equal true, @logger.error("Hello")
     end
 
+    test "with_level invokes its block one time, even when broadcasting to 2 or more loggers" do
+      invocations = 0
+      log1 = ::Logger.new(nil)
+      log2 = ::Logger.new(nil)
+
+      logger = BroadcastLogger.new(log1, log2)
+
+      invocations = 0
+      logger.with_level(:info) do
+        invocations += 1
+      end
+
+      assert_equal 1, invocations
+    end
+
+    test "with_level invokes its block one time, even when broadcasting to zero loggers" do
+      logger = BroadcastLogger.new()
+
+      invocations = 0
+      logger.with_level(:info) do
+        invocations += 1
+      end
+
+      assert_equal 1, invocations
+    end
+
+    test "with_level returns nil" do
+      logger = BroadcastLogger.new(
+        ::Logger.new(StringIO.new),
+        ::Logger.new(StringIO.new)
+      )
+
+      assert_nil logger.with_level(:debug) { :foo }
```

Comment:
On `main` this test fails with:

```
Expected [:foo, :foo] to be nil.
```

The reason being, on `main`, `with_level` hits this line in `method_missing`:

```
          loggers.map { |logger| logger.send(name, ...) }
```